### PR TITLE
Problem: The example doesnt compile on windows

### PR DESCRIPTION
**Pull request by gabm**
Solution: add namespaces, because the MSVC gets confused with `State` being defined as a class - and as a typedef in the base class

further: M_PI is missing...

Results look fine on MSVC 14/ VS 2015

### CODE REVIEW

File: examples/Robot1/main.cpp [hunk 1]
```diff
@@ -9,6 +9,11 @@
 #include <random>
 #include <chrono>
 
+#ifdef _WIN32
```
Comment (mherb) [inline examples/Robot1/main.cpp]: Wouldn't`#include <cmath>` help here? if not, at least chaning to `#ifndef M_PI` would be preferred here..

**Comment by kuepe-sl** [comment]
`#define _USE_MATH_DEFINES` must be the first line in the file, else you risk that it might not work. (see https://stackoverflow.com/a/6563891)  

The tricky thing here is, that `cmath` includes `math.h` for functions and defines, but it is included only once. `math.h` on the other side has separate header guards for its constants. So you can include it twice - once with functions only and another time for constants after defining `_USE_MATH_DEFINES`.

tl;dr this does not work:
```cpp
#include <cmath> // might be included indirectly by some header file
#define _USE_MATH_DEFINES
#include <cmath> // <-- won't define M_PI (cmath was already included)
```
but this works:
```cpp
#include <cmath> // might be included indirectly by some header file
#define _USE_MATH_DEFINES
#include <math.h> // <-- will define M_PI due to special header guards for constants
```

**Comment by mherb** [comment]
Thanks for the detailed comment @kuepe-sl, can you update the PR accordingly @gabm ?

**Comment by gabm** [comment]
done.... @mherb do u want to merge into develop or master?
